src/main/scala/org/opennetworkinsight/ProxyPostLDA.scala

```diff
@@ -0,0 +1,126 @@
+package org.opennetworkinsight
+
+import org.apache.log4j.{Level, Logger => ApacheLogger}
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.SparkContext
+import org.slf4j.{Logger, LoggerFactory}
+
+/**
+  *
+  */
+object ProxyPostLDA {
+
+  def getColumnNames(input: Array[String], sep: Char = ','): scala.collection.mutable.Map[String, Int] = {
+    val columns = scala.collection.mutable.Map[String, Int]()
+    val header = input.zipWithIndex
+    header.foreach(tuple => columns(tuple._1) = tuple._2)
+    columns
+  }
+
+  def proxyPostLDA(inputPath: String, resultsFilePath: String, threshold: Double, documentResults: Array[String],
+                 wordResults: Array[String], sc: SparkContext, sqlContext: SQLContext, logger: Logger) = {
+
+    val logger = LoggerFactory.getLogger(this.getClass)
+    ApacheLogger.getLogger("org").setLevel(Level.OFF)
+    ApacheLogger.getLogger("akka").setLevel(Level.OFF)
+
+    logger.info("Proxy post LDA starts")
+
+    val topics_lines = documentResults
+    val words_lines = wordResults
+
+    val l_topics = topics_lines.map(line => {
+      val ip = line.split(",")(0)
+      val topicProbs = line.split(",")(1).split(' ').map(_.toDouble)
+      (ip, topicProbs)
+    }).map({case (ip, topicProbs) => ip -> topicProbs }).toMap
+
+    val topics = sc.broadcast(l_topics)
+
+    val l_words  = words_lines.map(line => {
+      val word = line.split(",")(0)
+      val probPerTopic = line.split(",")(1).split(' ').map(_.toDouble)
+      (word, probPerTopic)
+    }).map({case (word, probPerTopic)  => word -> probPerTopic}).toMap
+
+    val words = sc.broadcast(l_words)
+
+    logger.info("Proxy post LDA starts")
+
+    var df_cols = new Array[String](0)
+
+    val multidata = {
+      var df = sqlContext.parquetFile(inputPath.split(",")(0)).filter("proxy_date is not null and proxy_time is not null and proxy_clientip is not null")
+      val files = inputPath.split(",")
+      for ((file, index) <- files.zipWithIndex) {
+        if (index > 1) {
+          df = df.unionAll(sqlContext.parquetFile(file).filter("proxy_date is not null and proxy_time is not null and proxy_clientip is not null"))
+        }
+      }
+      df = df.select("proxy_date",
```

Comment:
I don't like how "scoring" is pulling in columns for OA, but that's how it is right now... the trick is joining the scored records with the values in the other columns... I don't think there's a good unique key so it might be tricky to separate scoring from pulling data for OA (even though that's how I think it should be)

just something to think about